### ISSUE TITLE
Closure inside IF statement is not tokenized correctly

### ISSUE THREAD

**Issue by pbull** [squizlabs/PHP_CodeSniffer#527]
When running PHPCS using PSR2 standards, the following use of `array_filter` with an anonymous function appears to follow standards and PHPCS does not throw any errors:
```
$commandName = "test";
$commands = array();

$test = array_filter(
    $commands,
    function ($cmd) use ($commandName) {
        return ($cmd['name'] == $commandName);
    }
);
```

When I use that same code in an `if` statement, the following errors are reported:
* Expected 0 spaces before opening parenthesis; 1 found (Squiz.Functions.MultiLineFunctionDeclaration.SpaceBeforeOpenParen)
* Multi-line function call not indented correctly; expected 4 spaces but found 8 (PSR2.Methods.FunctionCallSignature.Indent)
```
$commandName = "test";
$commands = array();

if (array_filter(
    $commands,
    function ($cmd) use ($commandName) {
        return ($cmd['name'] == $commandName);
    }
)) {
    // Do something
}
```
In attempting to debug the situation, it seems that the `SpaceBeforeOpenParen` error is being triggered by the space between "function" and the opening paren. When I remove that space, the `SpaceBeforeOpenParen` goes away but then of course it throws a `SpaceAfterFunction` error.

**Comment by aik099** [comment]
I'm hoping code in example is not even close to production website, but I agree, that this might be a bug in sniff itself.

**Comment by pbull** [comment]
Code is an example for the purpose of demonstrating the errors I ran into when running PHP_Codesniffer, but I appreciate the very helpful code review.

**Comment by gsherwood** [comment]
The tokenizer was assuming nothing important happens within the conditions of an IF statement. But it no longer assumes that. Thanks for reporting.
